### ISSUE TITLE
Make the IEnumerable max depth for actual and expected output configurable

### ISSUE THREAD

**Issue by andre-ss6** [xunit/xunit#2469]
Currently, the max depth for printing actual and expected IEnumerables is hardcoded as 5, even at v3 (though v3 is at least referencing the const fields).

It's kinda frustrating to have to debug a thery every time in order to know which specific test failed.

**Comment by andre-ss6** [comment]
I'm up for contributing with this, I just need to know the design decisions (will it be configurable programmatically or via configuration file? or both? or an overload for Asserts, perhaps?).

**Comment by bradwilson** [comment]
Yeah, I've been thinking about how to do this, since there's actually three separate copies of `ArgumentFormatter` (which is what's responsible for doing this): in `xunit.v3.assert`, in `xunit.v3.common`, and in `xunit.v3.runner.common`. They don't share a single instance, since 

I think I'm going to end up using environment variables which get set by the in-process runner. That way every instance can just pick it up ambiently.

**Comment by bradwilson** [comment]
Available in v3 `1.1.0-pre.2` https://xunit.net/docs/using-ci-builds

Documentation has not been updated yet, but there are several ways to configure this now. The default value in all cases is `5`, and setting the value to `0` will always print the whole enumeration.

1. If using `xunit.v3.runner.console`, add `-printMaxEnumerableLength 123`
1. If using `dotnet run` with native xUnit.net UX, add `-printMaxEnumerableLength 123`
1. If using `dotnet run` with the Microsoft Testing Platform UX, add `--print-max-enumerable-length 123`
1. If using `dotnet test` in VSTest mode, add `-- xUnit.PrintMaxEnumerableLength=123`
1. If using `dotnet test` in Microsoft Testing Platform mode, add `-- --print-max-enumerable-length 123`
1. Runner-independent configuration via `xunit.runner.json`:
   ```json
   {
     "printMaxEnumerableLength": 123
   }
   ```
1. Configure via environment variable: `XUNIT_PRINT_MAX_ENUMERABLE_LENGTH=123`

Examples of option 2 above:

![Image](https://github.com/user-attachments/assets/4d67ea6b-ef1c-4d12-a4df-2c94e14586bc)
